hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/BlockManagerImpl.java

```diff
@@ -121,7 +121,7 @@ public long persistPutBlock(KeyValueContainer container,
       // default blockCommitSequenceId for any block is 0. It the putBlock
       // request is not coming via Ratis(for test scenarios), it will be 0.
       // In such cases, we should overwrite the block as well
-      if ((bcsId != 0) && (bcsId <= containerBCSId)) {
+      if ((bcsId != 0) && (bcsId < containerBCSId)) {
```

Comment:
Looks like this will still fail the write if we try to add a block whose BCSID is less than that in the container. Can you check the tests to see if this is being handled?